test-util/src/main/java/care/smith/fts/test/FhirGenerator.java

```diff
@@ -64,83 +59,27 @@ public Bundle generateBundle(int totalEntries, int pageSize) {
    */
   public String generateString() {
     String s = templateBuffer.toString();
-    for (Map.Entry<String, Replacement> m : inputReplacements.entrySet()) {
-      var replacement = m.getValue().apply();
-      addToReplacementMap(m, replacement);
-      s = s.replace(m.getKey(), replacement);
+    for (Map.Entry<String, Supplier<String>> m : inputReplacements.entrySet()) {
+      s = s.replace(m.getKey(), m.getValue().get());
     }
     return s;
   }
 
-  private void addToReplacementMap(Entry<String, Replacement> m, String replacement) {
-    var n = replacements.getOrDefault(m.getKey(), new ArrayList<>());
-    n.add(replacement);
-    replacements.put(m.getKey(), n);
-  }
-
-  public InputStream generateInputStream() {
-    return new ByteArrayInputStream(generateString().getBytes());
-  }
-
-  public interface Replacement {
-    String apply();
-  }
-
-  public static class UUID implements Replacement {
-    @Override
-    public String apply() {
-      return java.util.UUID.randomUUID().toString();
-    }
-  }
-
-  public static class Fixed implements Replacement {
-    private final String value;
-
-    public Fixed(String value) {
-      this.value = value;
-    }
-
-    @Override
-    public String apply() {
-      return value;
-    }
-  }
-
-  public static class OneOfList implements Replacement {
-    private final List<String> values;
-    private final Random random = new Random();
-
-    public OneOfList(List<String> values) {
-      this.values = values;
-    }
-
-    public OneOfList(List<String> values, long seed) {
-      this.values = values;
-      random.setSeed(seed);
-    }
-
-    @Override
-    public String apply() {
-      int nextInt = random.nextInt(values.size());
-      return values.get(nextInt);
-    }
-  }
-
-  public static class Incrementing implements Replacement {
+  public static class Incrementing implements Supplier<String> {
     private final String prefix;
     private long index;
 
-    public Incrementing(String prefix, long index) {
+    private Incrementing(String prefix, long index) {
       this.prefix = prefix;
       this.index = index;
     }
 
-    public Incrementing(String prefix) {
-      this(prefix, 0);
+    public static Incrementing withPrefix(String prefix) {
+      return new Incrementing(prefix, 0);
     }
 
     @Override
-    public String apply() {
+    public String get() {
       String s = prefix + index;
       index += 1;
       return s;
```

Comment:
Does it make sense to keep the Incrementing class? 
It should be possible to replace it with e.g. Stream.iterate where Incrementing is used.